File: A-Difference_Max.go
package main

import (
	"fmt"
	
)

func main(){
	var a, b, c, d, ans int
	fmt.Scan(&a, &b)
	fmt.Scan(&c, &d)
	
	C := [4] int {a - c, a - d, b - c, b - d}
	ans = -100000000
	for i := 0; i < 4; i++ {
		if ans < C[i]{
			ans = C[i]
		}
	}
	fmt.Println(ans)
	
	

}

File: A-Difference_Max.cpp
#include <iostream>
#include <vector>
using namespace std;

int main(){
    int a, b, c, d;
    cin >> a >> b;
    cin >> c >> d;
    vector<int> C = {a - c, a - d, b - c, b - d};
    int ans = -10000000;

    for (int i = 0; i < 4; i++){
        if (ans < C.at(i)){
            ans = C.at(i);
        }
    }
    cout << ans << endl;

}